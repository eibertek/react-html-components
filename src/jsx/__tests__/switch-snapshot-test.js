import React from 'react';
import Switch from '../switch';
import renderer from 'react/lib/ReactTestRenderer';

const switchWithListeners = (
    <Switch required={true}
                 label="test label"
                 name="test_input"
                 id="test"
                 placeholder="RadioButton"
                 extraClass="anotherClass"
                 validate={true}
                 errorMessage="wrong"
                 successMessage="correct"
                 changeCallback={function(){}}
                 mouseEnterCallback={function(){}}
                 mouseLeaveCallback={function(){}}/>
);

const switchWithoutListeners = (
    <Switch required={true}
                 label="test label"
                 name="test_input"
                 checked={true}
                 id="test"/>
);

const switchWithValue = (
    <Switch label="test label"
                 name="test_input"
                 id="test"
                 checked={false}
                 value="value"/>
);

describe('Switch', () => {
    it('with listeners matches snapshot', () => {
        const component = renderer.create(switchWithListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('without listeners matches snapshot', () => {
        const component = renderer.create(switchWithoutListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with value matches snapshot', () => {
        const component = renderer.create(switchWithValue);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
