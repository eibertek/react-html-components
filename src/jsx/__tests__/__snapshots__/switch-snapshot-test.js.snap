exports[`Switch with listeners matches snapshot 1`] = `
<div
  className="switch">
  <label>
    Off
    <input
      className="validate anotherClass"
      disabled={false}
      id="test"
      name="test_input"
      onChange={[Function bound _onChange]}
      onMouseEnter={[Function bound _onMouseEnter]}
      onMouseLeave={[Function bound _onMouseLeave]}
      placeholder="RadioButton"
      required={true}
      type="checkbox" />
    <span
      className="lever" />
    On
  </label>
</div>
`;

exports[`Switch with value matches snapshot 1`] = `
<div
  className="switch">
  <label>
    Off
    <input
      className=""
      disabled={false}
      id="test"
      name="test_input"
      onChange={[Function bound _onChange]}
      onMouseEnter={[Function bound _onMouseEnter]}
      onMouseLeave={[Function bound _onMouseLeave]}
      placeholder=""
      required={false}
      type="checkbox" />
    <span
      className="lever" />
    On
  </label>
</div>
`;

exports[`Switch without listeners matches snapshot 1`] = `
<div
  className="switch">
  <label>
    Off
    <input
      className=""
      disabled={false}
      id="test"
      name="test_input"
      onChange={[Function bound _onChange]}
      onMouseEnter={[Function bound _onMouseEnter]}
      onMouseLeave={[Function bound _onMouseLeave]}
      placeholder=""
      required={true}
      type="checkbox" />
    <span
      className="lever" />
    On
  </label>
</div>
`;
